File: frontend/src/types/timesheet.ts
// Tipos específicos para Timesheet Analysis

export interface TimesheetRow {
  id: string;
  date: string;
  nome: string;
  error: string;
  team: string;
  corporation: string;
  payrate: string;
  add_time_hour: string;
  remove_time_hour: string;
  add_dollar: string;
  remove_dollar: string;
  total: string;
}

export interface TimesheetChartData {
  labels: string[];
  datasets: {
    label: string;
    data: number[];
    borderColor: string;
    backgroundColor: string;
    tension: number;
  }[];
}

export interface TimesheetTooltipExternalProps {
  tooltip: Record<string, unknown>;
  chartLabels: string[];
  chartDatasets: {
    label: string;
    data: number[];
    borderColor: string;
    backgroundColor: string;
    tension: number;
  }[];
  year: number;
  month: number;
  canvas?: HTMLCanvasElement;
  data: TimesheetRow[];
}

export interface JobCostingTimesheetRow {
  id: number;
  client: string;
  jobsite: string;
  lot_building: string | null;
  worktype: string | null;
  employee_name: string;
  regular_rate: number;
  regular_hours: number;
  reference_month: string;
  created_at: string;
} 

File: BOR2/apps/api/internal/domain/timesheet.go
package domain

import "time"

// TimesheetRow maps to the timesheet_analysis table.
type TimesheetRow struct {
	ID             string     `json:"id"`
	Date           *time.Time `json:"date,omitempty"`
	Nome           string     `json:"nome"`
	Error          string     `json:"error"`
	Team           string     `json:"team"`
	Corporation    string     `json:"corporation"`
	Payrate        float64    `json:"payrate"`
	AddTimeHour    float64    `json:"addTimeHour"`
	RemoveTimeHour float64    `json:"removeTimeHour"`
	AddDollar      float64    `json:"addDollar"`
	RemoveDollar   float64    `json:"removeDollar"`
	Total          float64    `json:"total"`
	Jobsite        string     `json:"jobsite"`
	LotBuilding    string     `json:"lotBuilding"`
	Worktype       string     `json:"worktype"`
	RegularHours   float64    `json:"regularHours"`
	CreatedAt      time.Time  `json:"createdAt"`
}

type TimesheetRowFilters struct {
	Team        string
	Corporation string
	Nome        string
}
